SleepEngine/Engine/GameTemplate/Scene.h

```diff
@@ -0,0 +1,23 @@
+// Copyright 2019 Tihran Katolikian
+
+#pragma once
+
+BEGIN_NAMESPACE_SLEEP
+
+class Scene
+{
+public:
+    using Initer = std::function<void()>;
+
+    Scene(Initer&& initer, std::string_view id);
+
+    std::string_view getID() const { return m_id; }
```

Comment:
macros? btw i ll refact them later)

Reply:
There is string -> string_view conversion so macros won't work properly)